pennylane/transforms/optimization/pattern_matching.py

```diff
@@ -244,28 +243,23 @@ def circuit():
                             node = group.template_dag.get_node(index)
                             inst = copy.deepcopy(node.op)
 
-                            inst._wires = Wires(wires)
-
-                            adjoint(apply, lazy=False)(inst)
+                            inst = qml.map_wires(inst, wire_map=dict(zip(inst.wires, wires)))
+                            adjoint(qml.apply, lazy=False)(inst)
 
                     # Add the unmatched gates.
                     for node_id in substitution.unmatched_list:
                         node = circuit_dag.get_node(node_id)
                         inst = copy.deepcopy(node.op)
-                        apply(inst)
+                        qml.apply(inst)
 
-                tape = tape_inside
+                tape = qml.map_wires(input=tape_inside, wire_map=inverse_wires_map)
 
     for op in tape.operations:
-        op._wires = Wires([inverse_wires_map[wire] for wire in op.wires.tolist()])
```

Comment:
Before we ALWAYS needed to use the `inverse_wires_map` because the `commutation_dag` mapped the wires of the `tape` object in-place.

Now, using `qml.map_wires` this doesn't happen, thus we don't need to re-map the wires.